tests/unhappypath/unhappypath_test.go

```diff
@@ -0,0 +1,281 @@
+		for _, anode := range node.GetWorkerNodes() {
+			// TODO need to remove tcpdump installation using yum
+			// once https://portworx.atlassian.net/browse/PTX-12174 is resolved.
+			logrus.Infof("installing tcpdump on node %s", anode.Name)
+			cmd := "yum install -y tcpdump"
+			_, err := Inst().N.RunCommandWithNoRetry(anode, cmd, node.ConnectionOpts{
+				Timeout:         1 * time.Minute,
+				TimeBeforeRetry: 5 * time.Second,
+				Sudo:            true,
+			})
+			if err != nil {
+				logrus.Warnf("failed to install tcpdump on node %s: %v", anode.Name, err)
+				break
+			}
+			//Check tcpdump command
+			cmd = "compgen -c | grep tcpdump"
+			out, errInSSH := Inst().N.RunCommandWithNoRetry(anode, cmd, node.ConnectionOpts{
+				Timeout:         1 * time.Minute,
+				TimeBeforeRetry: 5 * time.Second,
+				Sudo:            true,
+			})
+			if err != nil {
+				logrus.Warnf("failed to install tcpdump on node %s: %v", anode.Name, errInSSH)
+			}
+			if !strings.Contains(out, "tcpdump") {
+				logrus.Warnf("TCP dump is not avaiable in %s:", anode.Name)
+			}
+		}
+	}
+})
+
+// This test is to verify stability of the system when there  is  a network error on the system.
+var _ = Describe("{NetworkErrorInjection}", func() {
+	var testrailID = 3526435
+	injectionType := "drop"
+	//TODO need to fix this issue later.
+	// testrailID corresponds to: https://portworx.testrail.net/index.php?/cases/view/35264
+	var runID int
+	JustBeforeEach(func() {
+		runID = testrailuttils.AddRunsToMilestone(testrailID)
+	})
+	var contexts []*scheduler.Context
+	It("Inject network error while applications are running", func() {
+		contexts = make([]*scheduler.Context, 0)
+
+		for i := 0; i < Inst().GlobalScaleFactor; i++ {
+			logrus.Infof("Iteration number %d", i)
+			contexts = append(contexts, ScheduleApplications(fmt.Sprintf("applicationscaleup-%d", i))...)
+		}
+		currentTime := time.Now()
+		timeToExecuteTest := time.Now().Local().Add(time.Hour*time.Duration(totalTimeInHours) +
+			time.Minute*time.Duration(0) +
+			time.Second*time.Duration(0))
+
+		// Set Autofs trim
+		currNode := node.GetWorkerNodes()[0]
+		err := Inst().V.SetClusterOpts(currNode, map[string]string{
+			"--auto-fstrim": "on",
+		})
+		if err != nil {
+			err = fmt.Errorf("error while enabling auto fstrim, Error:%v", err)
+			Expect(err).NotTo(HaveOccurred())
+		}
+
+		Step("Verify applications after deployment", func() {
+			for _, ctx := range contexts {
+				ValidateContext(ctx)
+			}
+		})
+		Step("Create snapshot schedule policy", func() {
+			createSnapshotSchedule(contexts)
+		})
+
+		for int64(timeToExecuteTest.Sub(currentTime).Seconds()) > 0 {
+			// TODO core check
+			logrus.Infof("Remaining time to test in minutes : %d ", int64(timeToExecuteTest.Sub(currentTime).Seconds()/60))
+			Step("Set packet loss on random nodes ", func() {
+				//Get all nodes and set eth0
+				nodes := node.GetWorkerNodes()
+				numberOfNodes := int(math.Ceil(float64(0.40) * float64(len(nodes))))
```

Comment:
here a simple logic of:
```
numberOfNodes = len(nodes)*4/10
```

should work.

It will make it easier to read the code as well.